package/network/services/dropbear/files/dropbear.init

```diff
@@ -12,28 +12,44 @@ PIDCOUNT=0
 
 extra_command "killclients" "Kill ${NAME} processes except servers and yourself"
 
-_dropbearkey()
-{
-	/usr/bin/dropbearkey "$@" 0<&- 1>&- 2>&-
-}
+_dropbearkey() { /usr/bin/dropbearkey "$@" 0<&- 1>&- 2>&-; }
+
+dumb_stat() { ls -Lln "$1" | tr -s '\t ' ' ' ; }
+dumb_stat_perm() { dumb_stat "$1" | cut -d ' ' -f 1 ; }
+dumb_stat_owner() { dumb_stat "$1" | cut -d ' ' -f 3 ; }
 
 # $1 - host key file name
-hk_verify()
+file_verify()
 {
 	[ -f "$1" ] || return 1
-	[ -s "$1" ] || return 2
-	_dropbearkey -y -f "$1" || return 3
+	# checking file ownership
+	[ "$(dumb_stat_owner "$1")" = "0" ] || {
+		chown 0 "$1"
+		[ "$(dumb_stat_owner "$1")" = "0" ] || return 2
+	}
+	# checking file permissions
+	[ "$(dumb_stat_perm "$1")" = "-rw-------" ] || {
+		chmod 0600 "$1"
+		[ "$(dumb_stat_perm "$1")" = "-rw-------" ] || return 3
+	}
+	# file is host key or not?
+	[ "$2" = "nokey" ] && return 0
+	# checking file contents (finally)
+	[ -s "$1" ] || return 4
+	_dropbearkey -y -f "$1" || return 5
```

Comment:
Please extend the commit description why you are doing these changes.

Reply:
Files in /etc/dropbear/ should be owned by root and have 0600 permissions to fulfil dropbear itself requirements.